python/pyspark/sql/types.py

```diff
@@ -1665,7 +1665,35 @@ def convert(self, obj, gateway_client):
         t.setNanos(obj.microsecond * 1000)
         return t
 
+
+class DatetimeNTZConverter(object):
+    def can_convert(self, obj):
+        def prefer_timestamp_ntz():
+            from pyspark.sql import SparkSession
+
+            session = SparkSession._activeSession
+
+            with SparkContext._lock:
+                context = SparkContext._active_spark_context
+                if context is not None and session is not None:
+                    return session._is_timestamp_ntz_preferred()
+                else:
+                    return False
+
+        return isinstance(obj, datetime.datetime) and obj.tzinfo is None and prefer_timestamp_ntz()
+
+    def convert(self, obj, gateway_client):
+        from pyspark import SparkContext
+
+        seconds = calendar.timegm(obj.utctimetuple())
+        jvm = SparkContext._jvm
+        return jvm.org.apache.spark.sql.catalyst.util.DateTimeUtils.microsToLocalDateTime(
+            int(seconds) * 1000000 + obj.microsecond
```

Comment:
Don't you afraid to overflow int?

Reply:
Python one doesn't have overflow :). This is also consistent with other places of conversion in Python.